content/posts/irods-development-update-january-2023.md

```diff
@@ -0,0 +1,240 @@
+Title: iRODS Development Update: January 2023
+Date: 2023-01-23 12:00
+Author: Kory Draughn
+Slug: irods-development-update-january-2023
+Status: published
+
+
+Happy New Year!
+
+It's 2023 and what better way to kick-start the new year than with a development update. So, let's jump right in.
+
+We've spent a good amount of time fixing memory issues in the iRODS 4.3.0 server. All of that work is now merged and the server is more stable for it. We've also been working to improve thread-safety within the server.
+
+If you compile iRODS from source frequently, then you'll be happy to know that support for [Ccache](https://ccache.dev) has been added to the iRODS Development Environment. This enhancement has significantly improved compile-times. Iterating on the server has never been easier.
+
+We have an update about the GenQuery reimplementation. We've decided to ship it as a separate package. It will be maintained in its own repository so that we can get it in the hands of the community early. Doing that allows us to deploy bug fixes and enhancements frequently because it won't be tied to a particular version of the iRODS. We'll release more information as development moves forward. If that sounds interesting, then please get involved. We would love to hear feedback. You can follow along at [https://github.com/irods/irods_api_plugin_genquery2](https://github.com/irods/irods_api_plugin_genquery2).
+
+The iRODS Testing Environment now allows SSL to be enabled for federation testing. We've also added an experimental xunit viewer for viewing test results.
+
+The iCommands are receiving some improvements. `iinit` will soon provide options that instruct it to prompt for SSL and authentication configuration properties. Handling of long path strings is being improved as well. `itree` is being updated to match the behavior of the Linux `tree` command more closely.
+
+Resource objects obtained via the Python iRODS Client (PRC) now expose the resource parent ID, resource parent name, and resource hierarchy. Compatibility with Microsoft Windows has been improved in the PRC as well.
+
+We've tested and verified that the S3 Resource Plugin for iRODS 4.2.11 and 4.3.0 is compatible with Wasabi S3.
```

Comment:
could/should link to https://wasabi.com/ ?

Reply:
Done.